qiskit/utils/quantum_instance.py

```diff
@@ -505,9 +512,9 @@ def execute(self, circuits, had_transpiled: bool = False):
             # transpile here, the method always returns a copied list
             circuits = self.transpile(circuits)
 
-        from qiskit.providers import BackendV1
+        from qiskit.providers import Backend
 
-        circuit_job = isinstance(self._backend, BackendV1)
+        circuit_job = isinstance(self._backend, Backend)
         if self.is_statevector and self._backend.name() == "aer_simulator_statevector":
```

Comment:
Is it possible that `self._backend` is either V1 or V2 here? Then we need to check whether to call `_backend.name` or `_backend.name()`.

Reply:
Yeah, that's a good call I was just being lazy because if it's `BackendV2` then it's not Aer yet because Aer hasn't migrated to backendv2 yet. So `is_statevector` will only return `True` in practice on `BackendV1`. But this is just waiting for a bug on a future aer release when it switches to `BackendV2` so I'll fix this.

Reply:
Done in: https://github.com/Qiskit/qiskit-terra/pull/7563/commits/09d60907576fddc798ee7897513d90d8ed1852f3

Reply:
Maybe I'm confused, but doesn't 09d6090 now hardcode the correct call for `BackendV2` so that it will be broken with the current `Aer` ?

Reply:
It's using the `QuantumInstance`'s `backend_name` property defined here: https://github.com/Qiskit/qiskit-terra/pull/7563/files/de32972d00c9588a45c290349517bc67dbd61b08#diff-d65336fff02fc3baf6d34babd7bb5596cb1bed2a3a6bce80046677d0217ddd60R975 which I updated in earlier commits handle either V1 or V2 based on the backend interface version

Reply:
Oh yeah, I misread your edit.